server/src/main/java/org/elasticsearch/index/mapper/DocumentParserContext.java

```diff
@@ -637,6 +637,13 @@ public DocumentDimensions getDimensions() {
         return dimensions;
     }
 
+    /**
+     * Indicates if the index is configured to use synthetic source.
+     */
+    public boolean isSourceSynthetic() {
+        return indexSettings().getMode().isSyntheticSourceEnabled() || mappingLookup.isSourceSynthetic();
```

Comment:
I wonder whether we can omit the `indexSettings().getMode().isSyntheticSourceEnabled()` part here? If `mappingLookup.isSourceSynthetic()` is `true` then `indexSettings().getMode().isSyntheticSourceEnabled()` should be evaluate `true`?

Reply:
Good point, this is not needed.